src/style-spec/expression/expression.js

```diff
@@ -10,7 +9,7 @@ export interface Expression {
     +type: Type;
 
     static parse(args: Array<mixed>, context: ParsingContext): ?Expression;
-    evaluate(ctx: EvaluationContext): Value;
+    evaluate(ctx: EvaluationContext): any;
```

Comment:
Why not `Value`?

Reply:
It forces every use of `evaluate` to cast the result through `any` to the actual concrete type. The right way to fix this is to make `Expression` a generic type on the result type, but I wanted to avoid getting into that for now.

Reply:
Ah, right.  Yeah we should try the generic type approach, but definitely later since there's a decent chance that effort will stumble on flow limitations, heh